docker/Dockerfile.x86_64-unknown-netbsd

```diff
@@ -13,10 +13,17 @@ RUN /xargo.sh
 COPY netbsd.sh /
 RUN /netbsd.sh
 
+COPY toolchain.cmake /opt/toolchain.cmake
+ENV CMAKE_TOOLCHAIN_FILE=/opt/toolchain.cmake
+ENV CROSS_CMAKE_OBJECT_FLAGS="-ffunction-sections -fdata-sections -fPIC -m64"
+
 ENV CROSS_TOOLCHAIN_PREFIX=x86_64-unknown-netbsd-
 ENV CROSS_SYSROOT=/usr/local/x86_64-unknown-netbsd
 ENV CARGO_TARGET_X86_64_UNKNOWN_NETBSD_LINKER="$CROSS_TOOLCHAIN_PREFIX"gcc \
     AR_x86_64_unknown_netbsd="$CROSS_TOOLCHAIN_PREFIX"ar \
     CC_x86_64_unknown_netbsd="$CROSS_TOOLCHAIN_PREFIX"gcc \
     CXX_x86_64_unknown_netbsd="$CROSS_TOOLCHAIN_PREFIX"g++ \
-    BINDGEN_EXTRA_CLANG_ARGS_x86_64_unknown_netbsd="--sysroot=$CROSS_SYSROOT"
+    BINDGEN_EXTRA_CLANG_ARGS_x86_64_unknown_netbsd="--sysroot=$CROSS_SYSROOT" \
+    CROSS_CMAKE_SYSTEM_NAME=NetBSD \
+    CROSS_CMAKE_SYSTEM_PROCESSOR=x86_64 \
+    CROSS_CMAKE_CRT=netbsd
```

Comment:
All of the BSDs use their own `libc` variants, so I've given them all different names.